tests/suites/deploy/deploy_charms.sh

```diff
@@ -222,6 +222,27 @@ run_deploy_lxd_to_container() {
 	destroy_model "${model_name}"
 }
 
+# Checks the install hook resolving with --no-retry flag
+run_resolve_charm() {
+	echo
+
+	model_name="test-resolve-charm"
+	file="${TEST_DIR}/${model_name}.log"
+
+	ensure "${model_name}" "${file}"
+
+	charm=./testcharms/charms/simple-resolve
+	juju deploy "${charm}"
+
+	wait_for "error" '.applications["simple-resolve"] | ."application-status".current'
+
+	juju resolve --no-retry simple-resolve/0
+
+	wait_for "active" '.applications["simple-resolve"] | ."application-status".current'
+
+	destroy_model "${model_name}"
+}
```

Comment:
I see that you have copied over `--no-retry` from the original test. function names in the test seem to imply that it is asserted in the test that the failed hook is not re-run, but this is never checked.

Worth evaluating the purpose of this test, since the original seems broken

Reply:
Do you mean to add a check that there was only ONE hook re-run (after the `juju resolve --no-retry` execution)?

Reply:
I've added a message `wait_for` ("No install hook") to check the `--no-retry` flag behavior.